mde-services/src/main/java/de/terrestris/mde/mde_backend/service/MetadataCollectionService.java

```diff
@@ -170,6 +177,24 @@ public String create(String title, String cloneMetadataId) throws IOException {
       return metadataId;
     }
 
+    private void updateLegendSize(JsonIsoMetadata data) {
+      try {
+        if (data.getServices() == null) {
+          return;
+        }
+        for (var service : data.getServices()) {
+          if (service.getLegendImage() != null && service.getLegendImage().getUrl() != null) {
+            var img = ImageIO.read(new URI(replaceValues(service.getLegendImage().getUrl())).toURL());
+            service.getLegendImage().setWidth(img.getWidth());
+            service.getLegendImage().setHeight(img.getHeight());
```

Comment:
Very minor and probably not desired, but instead of wrapping the loop in a try/catch, we could just wrap the loop block to proceed if a single replacement fails.

Reply:
Good point, done